### PR TITLE
Try to fix nullreff

### PR DESCRIPTION
**Pull request by Bond-009**
Try to fix https://github.com/jellyfin/jellyfin/commit/d217f1614e1fb93d1549ff4b7fad7bfbdcba6204#commitcomment-36764469

### CODE REVIEW

**Comment by PrplHaz4** [comment]
```
[20:44:29] [ERR] Error processing request
System.NullReferenceException: Object reference not set to an instance of an object.
   at MediaBrowser.Api.Playback.MediaInfoService.SetDeviceSpecificData(BaseItem item, MediaSourceInfo mediaSource, DeviceProfile profile, AuthorizationInfo auth, Nullable`1 maxBitrate, Int64 startTimeTicks, String mediaSourceId, Nullable`1 audioStreamIndex, Nullable`1 subtitleStreamIndex, Nullable`1 maxAudioChannels, String playSessionId, Guid userId, Boolean enableDirectPlay, Boolean forceDirectPlayRemoteMediaSource, Boolean enableDirectStream, Boolean enableTranscoding, Boolean allowVideoStreamCopy, Boolean allowAudioStreamCopy) in /opt/jellyfin/MediaBrowser.Api/Playback/MediaInfoService.cs:line 544
   at MediaBrowser.Api.Playback.MediaInfoService.SetDeviceSpecificData(Guid itemId, PlaybackInfoResponse result, DeviceProfile profile, AuthorizationInfo auth, Nullable`1 maxBitrate, Int64 startTimeTicks, String mediaSourceId, Nullable`1 audioStreamIndex, Nullable`1 subtitleStreamIndex, Nullable`1 maxAudioChannels, Guid userId, Boolean enableDirectPlay, Boolean forceDirectPlayRemoteMediaSource, Boolean enableDirectStream, Boolean enableTranscoding, Boolean allowVideoStreamCopy, Boolean allowAudioStreamCopy) in /opt/jellyfin/MediaBrowser.Api/Playback/MediaInfoService.cs:line 349
   at MediaBrowser.Api.Playback.MediaInfoService.GetPlaybackInfo(GetPostedPlaybackInfo request) in /opt/jellyfin/MediaBrowser.Api/Playback/MediaInfoService.cs:line 215
   at MediaBrowser.Api.Playback.MediaInfoService.Post(GetPostedPlaybackInfo request) in /opt/jellyfin/MediaBrowser.Api/Playback/MediaInfoService.cs:line 262
   at Emby.Server.Implementations.Services.ServiceExecGeneral.GetTaskResult(Task task) in /opt/jellyfin/Emby.Server.Implementations/Services/ServiceExec.cs:line 108
   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost httpHost, IRequest httpReq, HttpResponse httpRes, ILogger logger, CancellationToken cancellationToken) in /opt/jellyfin/Emby.Server.Implementations/Services/ServiceHandler.cs:line 80
   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken) in /opt/jellyfin/Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:line 518
```

**Comment by PrplHaz4** [comment]
@Bond-009 definitely getting further now...but playback never starts...
this is with https://github.com/jellyfin/jellyfin/pull/2266/commits/84d1b12530b4c53910b300b040fe5f316523427e
```
lots of these repeating...
[h264 @ 0x55852a54dac0] no frame!
[NULL @ 0x55852b2c4000] SPS unavailable in decode_picture_timing
[NULL @ 0x55852b2c4000] non-existing PPS 0 referenced
[h264 @ 0x55852b2bf640] SPS unavailable in decode_picture_timing
[h264 @ 0x55852b2bf640] non-existing PPS 0 referenced
[h264 @ 0x55852b2bf640] decode_slice_header error
[h264 @ 0x55852b2bf640] no frame!
Unsupported codec with id 100358 for input stream 2
Unsupported codec with id 100358 for input stream 5
Unsupported codec with id 100358 for input stream 8
Unsupported codec with id 100358 for input stream 11
Unsupported codec with id 100358 for input stream 14
Unsupported codec with id 100358 for input stream 17
Unsupported codec with id 100358 for input stream 20
Unsupported codec with id 100358 for input stream 23
Unsupported codec with id 100358 for input stream 26
Unsupported codec with id 100358 for input stream 29
[21:09:06] [ERR] Error probing live tv stream
System.Text.Json.JsonException: The JSON value could not be converted to System.Int32. Path: $.streams[0].start_pts | LineNumber: 27 | BytePositionInLine: 35.
 ---> System.FormatException: Either the JSON value is not in a supported format, or is out of bounds for an Int32.
   at System.Text.Json.Utf8JsonReader.GetInt32()
   at System.Text.Json.Serialization.Converters.JsonConverterInt32.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoNotNullable`4.OnRead(ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonPropertyInfo.Read(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& readStack, Utf8JsonReader& reader, Exception ex)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.GetMediaInfoInternal(String inputPath, String primaryPath, MediaProtocol protocol, Boolean extractChapters, String probeSizeArgument, Boolean isAudio, Nullable`1 videoType, Boolean forceEnableLogging, CancellationToken cancellationToken) in /opt/jellyfin/MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs:line 399
   at Emby.Server.Implementations.Library.LiveStreamHelper.AddMediaInfoWithProbe(MediaSourceInfo mediaSource, Boolean isAudio, String cacheKey, Boolean addProbeDelay, CancellationToken cancellationToken) in /opt/jellyfin/Emby.Server.Implementations/Library/LiveStreamHelper.cs:line 76
   at Emby.Server.Implementations.Library.MediaSourceManager.OpenLiveStreamInternal(LiveStreamRequest request, CancellationToken cancellationToken) in /opt/jellyfin/Emby.Server.Implementations/Library/MediaSourceManager.cs:line 500
[21:09:06] [INF] Live stream opened: {"Protocol":"Http","Id":"613fb0d111b90e02708fd87a9672753e","Path":"https://abclive1-lh.akamaihd.net/i/abc_live01@423395/master.m3u8","Type":"Default","IsRemote":true,"ReadAtNativeFramerate":false,"IgnoreDts":true,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":false,"SupportsDirectPlay":true,"IsInfiniteStream":true,"RequiresOpening":true,"RequiresClosing":true,"LiveStreamId":"a17c75760a04e99b68cf766e11316e1c_09efa0d56b934a82adec00a87b837fb0_613fb0d111b90e02708fd87a9672753e","RequiresLooping":false,"SupportsProbing":true,"MediaStreams":[{"VideoRange":"SDR","DisplayTitle":"","IsInterlaced":true,"BitRate":20000000,"IsDefault":false,"IsForced":false,"Type":"Video","Index":-1,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false},{"DisplayTitle":"","IsInterlaced":false,"IsDefault":false,"IsForced":false,"Type":"Audio","Index":-1,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false}],"MediaAttachments":[],"Formats":[],"Bitrate":20000000,"RequiredHttpHeaders":{"User-Agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.85 Safari/537.36"},"AnalyzeDurationMs":3000}
[21:09:06] [INF] User policy for jjh. EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
[21:09:06] [INF] Profile: Unknown Profile, Path: https://abclive1-lh.akamaihd.net/i/abc_live01@423395/master.m3u8, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
```

**Comment by PrplHaz4** [comment]
@Bond-009 Nice work!! m3u and HDHR tv streams both work with https://github.com/jellyfin/jellyfin/pull/2266/commits/93ab829df523c36a44ffa2f9caaf96505a2b2e5c merged into master.

Still getting those weird ffmpeg log entries, but it may not be related, and does not error out on the probe - media seems to play ok.
```
lots of these repeating...
h264 @ 0x55852a54dac0] no frame!
[NULL @ 0x55852b2c4000] SPS unavailable in decode_picture_timing
[NULL @ 0x55852b2c4000] non-existing PPS 0 referenced
[h264 @ 0x55852b2bf640] SPS unavailable in decode_picture_timing
[h264 @ 0x55852b2bf640] non-existing PPS 0 referenced
[h264 @ 0x55852b2bf640] decode_slice_header error
[h264 @ 0x55852b2bf640] no frame!
Unsupported codec with id 100358 for input stream 2
Unsupported codec with id 100358 for input stream 5
Unsupported codec with id 100358 for input stream 8
Unsupported codec with id 100358 for input stream 11
Unsupported codec with id 100358 for input stream 14
Unsupported codec with id 100358 for input stream 17
Unsupported codec with id 100358 for input stream 20
Unsupported codec with id 100358 for input stream 23
Unsupported codec with id 100358 for input stream 26
Unsupported codec with id 100358 for input stream 29
[23:03:32] [INF] Live tv media info probe took 14.9139556 seconds
```

**Comment by PrplHaz4** [comment]
@dkanada master is broken without this - prob needs a merge

File: MediaBrowser.Api/Playback/MediaInfoService.cs [hunk 1]
```diff
@@ -264,41 +264,26 @@ public async Task<object> Post(GetPostedPlaybackInfo request)
...
-                catch (Exception ex)
-                {
-                    mediaSources = new List<MediaSourceInfo>();
-                    Logger.LogError(ex, "Could not find media sources for item id {id}", id);
-                    // TODO PlaybackException ??
-                    //result.ErrorCode = ex.ErrorCode;
-                }
 
-                result.MediaSources = mediaSources.ToArray();
+                // TODO handle supportedLiveMediaTypes ?
```
Comment (dkanada) [inline MediaBrowser.Api/Playback/MediaInfoService.cs]:
```suggestion
                // TODO handle supportedLiveMediaTypes?
```

**Comment by PrplHaz4** [comment]
Looks like this needs to be rebased with the updated tests PR merged...

**Pull request merged**